src/connectors/snowflake/trulens/connectors/snowflake/utils/server_side_evaluation_artifacts.py

```diff
@@ -0,0 +1,221 @@
+                    'rich',
+                    'scikit-learn',
+                    'scipy',
+                    'snowflake-snowpark-python',
+                    'sqlalchemy',
+                    'tqdm',
+                    'typing_extensions'
+                )
+                IMPORTS = ({imports})
+                HANDLER = 'run'
+            AS
+                $$
+            import _snowflake
+            import trulens.providers.cortex.provider
+            from snowflake.sqlalchemy import URL
+            from trulens.core import TruSession
+            from trulens.core.schema.feedback import FeedbackRunLocation
+
+
+            def run(session):
+                # Set up sqlalchemy engine parameters.
+                conn = session.connection
+                engine_params = {{}}
+                engine_params["paramstyle"] = "qmark"
+                engine_params["creator"] = lambda: conn
+                database_args = {{"engine_params": engine_params}}
+                # Ensure any Cortex provider uses the only Snowflake connection allowed in this stored procedure.
+                trulens.providers.cortex.provider._SNOWFLAKE_STORED_PROCEDURE_CONNECTION = (
+                    conn
+                )
+                # Run deferred feedback evaluator.
+                db_url = URL(
+                    account="{self._account}",
+                    user="{self._user}",
+                    password="password",
+                    database="{self._database_name}",
+                    schema="{self._schema_name}",
+                    warehouse="{self._warehouse}",
+                    role="{self._role}",
+                )
+                session = TruSession(
+                    database_url=db_url,
+                    database_check_revision=False,  # TODO: check revision in the future?
+                    database_args=database_args,
+                )
+                session.start_evaluator(
+                    run_location=FeedbackRunLocation.SNOWFLAKE,
+                    return_when_done=True,
+                    disable_tqdm=True,
+                )
+                $$
+            """
+        )
+
+    def _set_up_wrapper_stored_procedure(self) -> None:
+        self._run_query(
+            f"""
+            CREATE PROCEDURE IF NOT EXISTS {_WRAPPER_STORED_PROCEDURE_NAME}()
+                RETURNS VARCHAR
+                LANGUAGE SQL
+            AS
+                $$
+            BEGIN
+                CALL {_STORED_PROCEDURE_NAME}();
+                -- The following noop insert is done just so that the stream will clear.
+                -- Currently, the only way for the stream to clear is for the data involved to be in a DML query.
+                INSERT INTO TRULENS_FEEDBACKS (
+                    SELECT *
+                    EXCLUDE (METADATA$ACTION, METADATA$ISUPDATE, METADATA$ROW_ID)
+                    FROM {_STREAM_NAME}
+                    WHERE FALSE
+                );
```

Comment:
because I want to run the deferred evaluator only if there are new feedbacks at all. Like if you aren't ingesting any feedbacks I don't want a warehouse runnign the deferred evaluator code constantly looking for any.